### PR TITLE
Update roadmap.md 2023-08-22

### PR DESCRIPTION
**Pull request by mapedraza**

### CODE REVIEW

File: doc/roadmap.md [hunk 1]
```diff
@@ -33,29 +33,32 @@ The following list of features are planned to be addressed in the short term,
...
 
 - Allow multiple types in entity to support UNE 178503 requirements ([#3638](https://github.com/telefonicaid/fiware-orion/issues/3638))
-- ~New subscripition modes (create only, update only, delete only and combinations)~ ([#1494](https://github.com/telefonicaid/fiware-orion/issues/1494)
 - Pattern/filter batch updates ([#2389](https://github.com/telefonicaid/fiware-orion/issues/2389))
 - Notification endpoint alias ([#3655](https://github.com/telefonicaid/fiware-orion/issues/3655))
 - Aggregation operations API ([#3816](https://github.com/telefonicaid/fiware-orion/issues/3816))
-- ~Custom notifications: simplifying sending JSON requests~ ([#2560](https://github.com/telefonicaid/fiware-orion/issues/2560))
-
+- Rework commands and deprecate registration API
+- MQTT Retain flag
```
Comment (fgalan) [inline doc/roadmap.md:40]:
```suggestion
- MQTT Retain flag [#4388](https://github.com/telefonicaid/fiware-orion/issues/4388)
```

In addition, I'd suggest to put this on top of the list (L35)

File: doc/roadmap.md [hunk 2]
```diff
@@ -33,29 +33,32 @@ The following list of features are planned to be addressed in the short term,
...
 ## Medium term
 
 The following list of features are planned to be addressed in the medium term,
 typically within the subsequent release(s) generated in the next **9 months**
 after next planned release:
 
-
-- Advanced query language
+- Advanced query language ([4395](https://github.com/telefonicaid/fiware-orion/issues/4395))
 - Signed entities
```
Comment (fgalan) [inline doc/roadmap.md:49]: Create placeholder issue associated to this.

File: doc/roadmap.md [hunk 3]
```diff
@@ -33,29 +33,32 @@ The following list of features are planned to be addressed in the short term,
 and incorporated into the coming release(s) of the product:
 
 - Allow multiple types in entity to support UNE 178503 requirements ([#3638](https://github.com/telefonicaid/fiware-orion/issues/3638))
-- ~New subscripition modes (create only, update only, delete only and combinations)~ ([#1494](https://github.com/telefonicaid/fiware-orion/issues/1494)
 - Pattern/filter batch updates ([#2389](https://github.com/telefonicaid/fiware-orion/issues/2389))
 - Notification endpoint alias ([#3655](https://github.com/telefonicaid/fiware-orion/issues/3655))
 - Aggregation operations API ([#3816](https://github.com/telefonicaid/fiware-orion/issues/3816))
-- ~Custom notifications: simplifying sending JSON requests~ ([#2560](https://github.com/telefonicaid/fiware-orion/issues/2560))
-
+- Rework commands and deprecate registration API
```
Comment (fgalan) [inline doc/roadmap.md:39]:
```suggestion
- Rework commands (and deprecate registration API)
```

In addition, create placeholder issue associated to this.

File: doc/roadmap.md [hunk 4]
```diff
@@ -33,29 +33,32 @@ The following list of features are planned to be addressed in the short term,
...
 ([#3815](https://github.com/telefonicaid/fiware-orion/issues/3815))
-- Rework commands
 - Service provisioning API (pools, etc.) 
 (based in [#3843](https://github.com/telefonicaid/fiware-orion/issues/3843))
+- Remove registration API
+- Advanced subscription management ([4392](https://github.com/telefonicaid/fiware-orion/issues/4392))
+   - Custom ID subscription
+   - Query Subscrition by fields (endpoint, attributes...)
+   - Configurable statistics consolidation flag (to debug subscriptions) / Debug mode + last 10 subs status?
+   - Flag to disable metadata in subscriptions
```
Comment (fgalan) [inline doc/roadmap.md:61]:
```suggestion
- Advanced subscription management
   - Subscription debug mode (precise statistics consolidation, keep recent history of notifications sent, etc.)
   - Custom ID subscription
   - Query Subscrition by fields (endpoint, attributes...) ([4392](https://github.com/telefonicaid/fiware-orion/issues/4392))
   - Flag to disable metadata in subscriptions
- Remove registration API
```

Create placeholder issues for the 3 sub-items under "Advanced subscriptions management".

File: doc/roadmap.md [hunk 5]
```diff
@@ -71,6 +74,10 @@ you wish to get involved in the implementation or influence the roadmap
 
 The following list contains all features that were in the roadmap and have already been implemented.
 
+- ~New subscripition modes (create only, update only, delete only and combinations)~ ([#1494](https://github.com/telefonicaid/fiware-orion/issues/1494)
+- ~Custom notifications: simplifying sending JSON requests~ ([#2560](https://github.com/telefonicaid/fiware-orion/issues/2560))
+
+
```
Comment (fgalan) [inline doc/roadmap.md:80]: Fix code style and include link to the release which solved each issue.

File: doc/roadmap.md (hunk 1 above)
Comment (mapedraza) [reply doc/roadmap.md]: Done in https://github.com/telefonicaid/fiware-orion/pull/4396/commits/8a566a9c9fa3cc37f8810cf2f9d117334f6b69ae

File: doc/roadmap.md (hunk 2 above)
Comment (mapedraza) [reply doc/roadmap.md]: Done in https://github.com/telefonicaid/fiware-orion/pull/4396/commits/8a566a9c9fa3cc37f8810cf2f9d117334f6b69ae

File: doc/roadmap.md (hunk 4 above)
Comment (mapedraza) [reply doc/roadmap.md]: Done in https://github.com/telefonicaid/fiware-orion/pull/4396/commits/8a566a9c9fa3cc37f8810cf2f9d117334f6b69ae

File: doc/roadmap.md [hunk 6]
```diff
@@ -32,12 +32,12 @@ Disclaimer:
...
 and incorporated into the coming release(s) of the product:
 
+- MQTT Retain flag [#4388](https://github.com/telefonicaid/fiware-orion/issues/4388)
 - Allow multiple types in entity to support UNE 178503 requirements ([#3638](https://github.com/telefonicaid/fiware-orion/issues/3638))
-- ~New subscripition modes (create only, update only, delete only and combinations)~ ([#1494](https://github.com/telefonicaid/fiware-orion/issues/1494)
 - Pattern/filter batch updates ([#2389](https://github.com/telefonicaid/fiware-orion/issues/2389))
 - Notification endpoint alias ([#3655](https://github.com/telefonicaid/fiware-orion/issues/3655))
 - Aggregation operations API ([#3816](https://github.com/telefonicaid/fiware-orion/issues/3816))
-- ~Custom notifications: simplifying sending JSON requests~ ([#2560](https://github.com/telefonicaid/fiware-orion/issues/2560))
+- Rework commands (and deprecate registration API) ([4397](https://github.com/telefonicaid/fiware-orion/issues/4397))
```
Comment (fgalan) [inline doc/roadmap.md:40]:
```suggestion
- Rework commands (and deprecate registration API) ([#4397](https://github.com/telefonicaid/fiware-orion/issues/4397))
```

Just style... looking to existing entries.

File: doc/roadmap.md [hunk 7]
```diff
@@ -46,16 +46,20 @@ The following list of features are planned to be addressed in the medium term,
 typically within the subsequent release(s) generated in the next **9 months**
 after next planned release:
 
-
-- Advanced query language
-- Signed entities
+- Advanced query language ([4395](https://github.com/telefonicaid/fiware-orion/issues/4395))
+- Signed entities ([4398](https://github.com/telefonicaid/fiware-orion/issues/4398))
```
Comment (fgalan) [inline doc/roadmap.md:50]:
```suggestion
- Advanced query language ([#4395](https://github.com/telefonicaid/fiware-orion/issues/4395))
- Signed entities ([#4398](https://github.com/telefonicaid/fiware-orion/issues/4398))
```

Just style... looking to existing entries.

File: doc/roadmap.md (hunk 5 above)
Comment (mapedraza) [reply doc/roadmap.md]: Done in https://github.com/telefonicaid/fiware-orion/pull/4396/commits/1e36d4ffb68f9c776a7d72c3a2d15ce67f5bcc59

**Comment by mapedraza** [comment]
@fgalan This PR is ready to merge

**Pull request merged**